torch/nn/parallel/distributed.py

```diff
@@ -657,7 +660,19 @@ def forward(self, *inputs, **kwargs):
             self.reducer._set_forward_pass_work_handle(
                 work, self.ddp_uneven_inputs_config.ddp_join_divide_by_initial_world_size
             )
-
+            if self.ddp_uneven_inputs_config.ddp_join_throw_on_early_termination:
+                # Check if need to throw StopIteration.
+                zeros = torch.zeros(1, device=self.device)
+                # TODO: This allreduce can be scheduled and waited in the C++
```

Comment:
For something like this, we would also probably have to move throwing the `StopIteration` in the join logic from the beginning to the end of the join() logic so that all collective comm. still matches up in the right order